ts/features/walletV3/onboarding/screens/WalletOnboardingSelectPaymentMethodScreen.tsx

```diff
@@ -0,0 +1,98 @@
+import { useNavigation } from "@react-navigation/native";
+
+import I18n from "../../../../i18n";
+import {
+  WalletOnboardingRoutes,
+  WalletOnboardingStackNavigation
+} from "../navigation/navigator";
+import TopScreenComponent from "../../../../components/screens/TopScreenComponent";
+import WalletOnboardingPaymentMethodsList from "../components/WalletOnboardingPaymentMethodsList";
+import { PaymentMethodResponse } from "../../../../../definitions/pagopa/walletv3/PaymentMethodResponse";
+import { useOnFirstRender } from "../../../../utils/hooks/useOnFirstRender";
+import { walletGetPaymentMethods } from "../store/actions";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import {
+  isLoadingPaymentMethodsSelector,
+  walletOnboardingPaymentMethodsSelector
+} from "../store";
+import { OperationResultScreenContent } from "../../../../components/screens/OperationResultScreenContent";
+import { PaymentMethodStatusEnum } from "../../../../../definitions/pagopa/walletv3/PaymentMethodStatus";
+
+const WalletOnboardingSelectPaymentMethodScreen = () => {
+  const navigation = useNavigation<WalletOnboardingStackNavigation>();
+  const dispatch = useIODispatch();
+  const isLoadingPaymentMethods = useIOSelector(
+    isLoadingPaymentMethodsSelector
+  );
+  const paymentMethodsPot = useIOSelector(
+    walletOnboardingPaymentMethodsSelector
+  );
+  const availablePaymentMethods = pipe(
+    pot.getOrElse(
+      pot.map(paymentMethodsPot, el => el.paymentMethods),
+      null
+    ),
+    O.fromNullable,
+    O.map(el => el.filter(el => el.status === PaymentMethodStatusEnum.ENABLED)),
+    O.getOrElseW(() => [])
+  );
+
+  useOnFirstRender(() => {
+    dispatch(walletGetPaymentMethods.request());
+  });
+
+  const handleSelectedPaymentMethod = (
+    selectedPaymentMethod: PaymentMethodResponse
+  ) => {
+    navigation.navigate(WalletOnboardingRoutes.WALLET_ONBOARDING_START, {
+      paymentMethodId: selectedPaymentMethod.id
+    });
+  };
+
+  return (
+    <TopScreenComponent goBack>
+      {pot.isError(paymentMethodsPot) && (
+        <OperationResultScreenContent
+          pictogram="umbrellaNew"
+          title={I18n.t("genericError")}
+          subtitle={I18n.t("global.genericError")}
+          action={{
+            label: I18n.t("global.genericRetry"),
+            accessibilityLabel: I18n.t("global.genericRetry"),
+            onPress: () => dispatch(walletGetPaymentMethods.request())
+          }}
+        />
+      )}
+      {!pot.isError(paymentMethodsPot) && (
+        <SafeAreaView style={IOStyles.flex}>
+          <WalletOnboardingPaymentMethodsList
+            header={<PaymentMethodsHeading />}
+            isLoading={isLoadingPaymentMethods}
+            onSelectPaymentMethod={handleSelectedPaymentMethod}
+            paymentMethods={availablePaymentMethods}
+          />
+        </SafeAreaView>
+      )}
+    </TopScreenComponent>
```

Comment:
```suggestion
     <TopScreenComponent goBack>
      {pot.isError(paymentMethodsPot) ? (
        <OperationResultScreenContent
          pictogram="umbrellaNew"
          title={I18n.t("genericError")}
          subtitle={I18n.t("global.genericError")}
          action={{
            label: I18n.t("global.genericRetry"),
            accessibilityLabel: I18n.t("global.genericRetry"),
            onPress: () => dispatch(walletGetPaymentMethods.request())
          }}
        />
      ) : (
        <SafeAreaView style={IOStyles.flex}>
          <WalletOnboardingPaymentMethodsList
            header={<PaymentMethodsHeading />}
            isLoading={isLoadingPaymentMethods}
            onSelectPaymentMethod={handleSelectedPaymentMethod}
            paymentMethods={availablePaymentMethods}
          />
        </SafeAreaView>
      )}
    </TopScreenComponent
```

why double check?

Reply:
Just to make it more readable in case we will have other cases, but anyway has been addressed into cd9f233fec367a7b58b4d5b0555ede8d5b9b5d6f